### PR TITLE
[Database] Create Public Interfaces and Shared Data Types

### LINKED ISSUE #30: [Database] Create Public Interfaces and Shared Data Types
**What is the use case? Please describe.**
Create the public Java interface for each interface in the `edu.oakland.production.database` package, and the public methods for every data type in the `edu.oakland.helper.database` package. Note that this is part of sprint deliverable task 2.3.2.

**Setup**

To be done by @camrongrant1 , @kylepoterek , or @MathewYaldo before the rest of the team starts working on it:

1. Use the documentation at https://github.com/andrewdimmer/csi-3370-software-project/tree/database-production-uml-class-diagrams/docs/uml-class-diagrams/database to identify the classes and interfaces to create.
2. Edit this issue message to add a checkbox for each class or interface, and assign it to the correct team member.

**Steps**

To be done by each team member for each class or interface they are creating:

1. Checkout and/or pull the branch `database-public-interfaces`
2. Make the code changes to create the class or interface described in https://github.com/andrewdimmer/csi-3370-software-project/tree/database-production-uml-class-diagrams/docs/uml-class-diagrams/database
3. Test your changes locally with `gradle build` if you have Gradle installed, or test it with `./gradlew build` or `.\gradlew.bat build` if you do not.
4. Commit your changes with the message `[Database] Created <<<Class/Interface_Name_Here>>>`
5. Push your changes
6. If you are the first person to push, create a pull request draft with the name `[Database] Create Public Interfaces and Shared Data Types` and the rest of the information like description, tags, etc. For reference, the settings on the right hand side should match this issue, with the exception that the assignees on the PR are just those who are pushing code.
7. If you are the last person to push, release the pull request for reviews, and add @andrewdimmer as a reviewer.

**Tasks to Complete**

- [x] Assign Tasks - @camrongrant1 , @kylepoterek , or @MathewYaldo
- [x] Create DatabasePersistentStorage - @MathewYaldo and @letthatbe
- [x] Create DatabaseCommInterface - @Ian-Klaus
- [x] Create DatabaseCommManager - @Endritz  
- [x] Create DatabaseGisInterface - @robertsonbrinker
- [x] Create DatabaseGisManager - @Jake-wq

**Additional context**
Note: if for some reason you need to change a public interface or shared data type, reach out to @andrewdimmer who can help you manage making that BREAKING CHANGE.

### PR DESCRIPTION
**Pull request by MathewYaldo**
Created public interfaces

Closes #30

### CODE REVIEW

File: src/main/java/edu/oakland/production/database/DatabaseCommInterface.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package edu.oakland.production.database;
+
+import edu.oakland.helper.admin.LocationDataPoint;
+import edu.oakland.helper.admin.TrackData;
+
+public interface DatabaseCommInterface {
+
+  int receiveGetRfldRequest();
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/database/DatabaseCommInterface.java]:
```suggestion
  int receiveGetRfidRequest();
```

File: src/main/java/edu/oakland/production/database/DatabaseGisInterface.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import edu.oakland.helper.admin.LocationDataPoint;
+
+public interface DatabaseGisInterface {
+
+  public void receiveStoreRequest(LocationDataPoint locationDataPoint);
+
+  public String receiveModeRequest(String mode); 
+
+  public String receiveNextSatRequest(String currentStat);
```
Comment (andrewdimmer) [inline src/main/java/edu/oakland/production/database/DatabaseGisInterface.java]:
```suggestion
  public String receiveNextSatRequest(String currentSat);
```

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
Two really minor changes regarding spelling, then after that you should be good to go!

**Review by kylepoterek — APPROVED** [review]

**Review by andrewdimmer — APPROVED** [review]
Looks good to me! You're all set to merge!

**Pull request merged**